downgrade

```diff
@@ -448,12 +425,41 @@ parse_options() {
 }
 
 cli() {
-  parse_options "$@"
+  declare -a system_args
+
+  # get system and command-line arguments and parse everything
+  IFS=' ' read -r -a system_args <<<"$(read_downgrade_conf)"
```

Comment:
Can we add a test suite on this behavior? It's going to be tricky and I'd like to avoid regressions in the future.

I don't think we have any options that accept arguments with spaces in them right now, but if it's not too complicated, I think we should try to write this code to handle that now. In other words:

```conf
# Some header comment

# --example option
--actual "Option with spaces"
```

Should return `["--actual", "Option with spaces"]`. I think the code you have would return `["--actual", "Option", "with", "spaces"]`, right?

Reply:
Good catch! Yes this could be problematic and I will add a test suite for this later on.

Indeed, the code as of now will greedily split on spaces. Any suggestions btw on how to allow the splits to respect a clearly demarcated string? I will look into this further as well.

Reply:
Used two tricks to solve this.

1. Replace `xargs` with `tr` to ensure any quotes or demarcations stay intact:

https://github.com/pbrisbin/downgrade/blob/e99d30aad608e9412e72ddffea2e792d92ef1b04/downgrade#L38

2. Use standard variable expansions to parse the single-line string extracted from the configuration file:

https://github.com/pbrisbin/downgrade/blob/e99d30aad608e9412e72ddffea2e792d92ef1b04/downgrade#L439

Reply:
Let me know if these fixes make sense to you. The `eval` statement is a bit unusual but I can't think of anything else at the moment.

Reply:
Added a test suite for this in 503b4389888b777d02c25f32c3934432fca6ca66